metadeploy/api/serializers.py

```diff
@@ -481,10 +481,26 @@ def _validate_results(self, data):
 
     def validate(self, data):
         user = self._get_from_data_or_instance(data, "user")
-        org_id = self._get_from_data_or_instance(data, "org_id") or user.org_id
+        org_id = self._get_from_data_or_instance(data, "org_id") or getattr(
+            user, "org_id", None
+        )
         plan = self._get_from_data_or_instance(data, "plan")
         steps = self._get_from_data_or_instance(data, "steps", default=[])
 
+        scratch_org = None
+        if not org_id:
+            scratch_org_id = self.context["request"].session.get("scratch_org_id", None)
+            if scratch_org_id:
+                scratch_org = ScratchOrg.objects.filter(
+                    uuid=scratch_org_id, status=ScratchOrg.Status.complete
+                ).first()
+
+        if scratch_org:
+            org_id = scratch_org.org_id
+
+        if not org_id:
+            raise serializers.ValidationError(_("No valid org."))
```

Comment:
I'm not crazy about tracing the routes through this sequence of conditionals; I'll see if I can flatten it at all.